### PR TITLE
Support missing message fields

### PR DESCRIPTION
**Pull request by Doc94**
**Description:** Its a WIP about little changes in discord messages (WIP Because its working but not document) and anothe changes for discord features.

Reference: Tests in servers and https://github.com/discordapp/discord-api-docs/pull/1051

### CODE REVIEW

**Comment by Doc94** [comment]
I have a little question about add support to "message_reference" in the new type of message (follow channel) this param show the origin of follow.
Example using Discord API.
`[
    {
        "attachments": [],
        "tts": false,
        "embeds": [],
        "timestamp": "2019-08-09T15:45:32.088000+00:00",
        "mention_everyone": false,
        "id": "609411953697357825",
        "pinned": false,
        "edited_timestamp": null,
        "message_reference": {
            "channel_id": "326437277217849345",
            "guild_id": "322850917248663552"
        },
        "author": {
            "username": "Doc",
            "discriminator": "4444",
            "id": "216316781483130880",
            "avatar": "9e56b7acf4f1dafa7b75d5aec4b2deef"
        },
        "mention_roles": [],
        "content": "Official Fortnite #twitter-feed",
        "channel_id": "609411844272029716",
        "mentions": [],
        "type": 12
    }
]`

File: core/src/main/java/discord4j/core/object/entity/Message.java [hunk 1]
```diff
@@ -477,7 +477,10 @@ public int hashCode() {
         USER_PREMIUM_GUILD_SUBSCRIPTION_TIER_2(10),
 
         /** A message created when an user boost a guild and the guild reach the tier 3. */
-        USER_PREMIUM_GUILD_SUBSCRIPTION_TIER_3(11);
+        USER_PREMIUM_GUILD_SUBSCRIPTION_TIER_3(11),
+
+        /** A message created when an user follow a channel from another guild into specific channel (<a href="https://support.discordapp.com/hc/en-us/articles/360028384531-Server-Following-FAQ">Server Following</a>). */
+        CHANNEL_FOLLOW_ADD(11);
```
Comment (Shadorc) [inline core/src/main/java/discord4j/core/object/entity/Message.java]: I think it should be 12

Comment (Doc94) [reply core/src/main/java/discord4j/core/object/entity/Message.java]: oh.....
i know i forget something.

**Comment by Doc94** [comment]
Of commit bf1e0d7 im not secure about the flags, with integer optionals.
In case of messagereference im create this out of Message (dont know if this fine).
(Im try to create a class like Type for the 3 flags but dont know how to work the "<<")

**Comment by Doc94** [comment]
A example of Message::toString with flags https://hasteb.in/yavafoxe.kotlin

File: core/src/main/java/discord4j/core/object/entity/Guild.java [hunk 2]
```diff
@@ -398,6 +398,8 @@ public ContentFilterLevel getContentFilterLevel() {
 
     /**
      * Gets the enabled guild features.
+     * <br>
+     * You can see the avaible <a href="https://discordapp.com/developers/docs/resources/guild#guild-object-guild-features">guild features</a>
```
Comment (Shadorc) [inline core/src/main/java/discord4j/core/object/entity/Guild.java]: the avaible

**Comment by Doc94** [comment]
Fixed supress method, added a "deprecated" for future changes by discord (but dont think change this... maybe response with a message edited and not empty).

**Comment by quanticc** [comment]
I merged this while setting some API as Experimental.

Since this PR was a bit stale, @Doc94 Can you give us some feedback if you notice any issues? Thanks in advance.

**Pull request merged**